src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -342,15 +342,11 @@ fn check_terminator(
                 // some intrinsics are waved through if called inside the
                 // standard library. Users never need to call them directly
                 match tcx.fn_sig(def_id).abi() {
-                    abi::Abi::RustIntrinsic => match &tcx.item_name(def_id).as_str()[..] {
-                        | "size_of"
-                        | "min_align_of"
-                        | "needs_drop"
-                        => {},
-                        _ => return Err((
+                    abi::Abi::RustIntrinsic => if !is_intrinsic_whitelisted(tcx, def_id) {
```

Comment:
I'm very surprised this does not lead to any errors. This must mean something is wrong. You changed the behavior to always emit an error, even for using the `size_of` intrinsic in a min const fn.

Reply:
oh whoops, the `if` is after the `=>`. Nevermind